python/ray/_raylet.pyx

```diff
@@ -481,6 +482,58 @@ cdef raise_if_dependency_failed(arg):
         raise arg
 
 
+cdef c_bool determine_if_retryable(
+    Exception e,
+    const c_string serialized_retry_exception_allowlist,
+    FunctionDescriptor function_descriptor,
+):
+    """Determine if the provided exception is retryable, according to the
+    (possibly null) serialized predicate function.
+
+    If the serialized predicate function is an empty string or is None once
+    deserialized, the exception is considered retryable and we return True.
+
+    This method can raise exceptions in several cases:
+        - Deserialization fails (TypeError)
+        - Deserialized predicate function is not a function (TypeError)
+        - Predicate function raises an error (RuntimeError)
+        - Predicate function output is not a boolean (TypeError)
+    """
+    if len(serialized_retry_exception_allowlist) == 0:
+        # No exception allowlist specified, default to all retryable.
+        return True
```

Comment:
We should never see an empty string coming from the Python frontend, but this can still happen in tests and may be allowed in cross-language scenarios, so we add this to be defensive.